### PR TITLE
docs: add documentation and for default git hook containing DCO and c…

### PR DESCRIPTION
**Pull request by leongdl**
### What was the problem/requirement? (What/Why)
- We require developers and contributors to use Conventional Commits, DCO for all commit messages. This blocks on github PR review.
- We also require developers to run fmt, lint before submission to meet code standards.

### What was the solution? (How)
- Automate DCO and conventional commits with an example git commit hook. Added documentation how to use it. It is optional, but will help save developers time and energy.
- Added a new `hatch quality` target to run both fmt and lint together. I always run them back to back anyways.

### What is the impact of this change?
- Developer productivity!

### How was this change tested?
- I installed the git hook, and used it to generate this commit! 
- I tested with various git commit amends to make sure conventional commits is enforced.

### Was this change documented?

- Are relevant docstrings in the code base updated?
- Has the README.md been updated? If you modified CLI arguments, for instance.

### Does this PR introduce new dependencies?

*   [ ] This PR adds one or more new dependency Python packages. I acknowledge I have reviewed the considerations for adding dependencies in [DEVELOPMENT.md](https://github.com/aws-deadline/deadline-cloud/blob/mainline/DEVELOPMENT.md#dependencies).
*   [X] This PR does not add any new dependencies.

### Is this a breaking change?
No, docs only.

### Does this change impact security?
- No.

----

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -107,6 +109,58 @@ If you need change a commit message, then please see the
...
+This repository enforces the Developer Certificate of Origin (DCO) on Pull Requests. It requires all commit messages to contain the `Signed-off-by` line with an email address that matches the commit author.
+
+To generate the `Signed-off-by`, make sure `git commit` is passed `-s` option.
+
+### Automating Conventional Commits and DCO.
+
+To simplify Conventional Commit and DCO, it is recommended to setup a Git hook for the repository. Add the following as a git hook:
+
+- Edit `{repository}/.git/hooks/commit-msg`
+- `chmox +x `{repository}/.git/hooks/commit-msg`
```
Comment (jericht) [inline CONTRIBUTING.md]:
```suggestion
- `chmod +x {repository}/.git/hooks/commit-msg`
```

I'm assuming this is supposed to be chmod? Also an extra backtick in there

Comment (leongdl) [reply CONTRIBUTING.md]: Good one! Thanks!

**Review by godobyte — DISMISSED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -107,6 +109,58 @@ If you need change a commit message, then please see the
 [GitHub documentation on the topic](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/changing-a-commit-message)
 to guide you.
 
+### Digital Certificate of Origin.
+
+This repository enforces the Developer Certificate of Origin (DCO) on Pull Requests. It requires all commit messages to contain the `Signed-off-by` line with an email address that matches the commit author.
+
+To generate the `Signed-off-by`, make sure `git commit` is passed `-s` option.
```
Comment (AWS-Samuel) [inline CONTRIBUTING.md]:
```suggestion
To generate the `Signed-off-by` line, make sure `git commit` is passed with the `-s` flag.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -107,6 +109,58 @@ If you need change a commit message, then please see the
...
+
+### Automating Conventional Commits and DCO.
+
+To simplify Conventional Commit and DCO, it is recommended to setup a Git hook for the repository. Add the following as a git hook:
+
+- Edit `{repository}/.git/hooks/commit-msg`
+- `chmox +x {repository}/.git/hooks/commit-msg`
```
Comment (AWS-Samuel) [inline CONTRIBUTING.md]:
```suggestion
- `chmod +x {repository}/.git/hooks/commit-msg`
```

**Review by YutongLi291 — DISMISSED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (leongdl) [reply CONTRIBUTING.md]: Great suggestion! I'll change it.

File: CONTRIBUTING.md (hunk 3 above)
Comment (leongdl) [reply CONTRIBUTING.md]: Darn, I thought I commited Jericho's suggestion.

**Review by AWS-Samuel — APPROVED** [review]

**Review by jericht — APPROVED** [review]

**Review by godobyte — APPROVED** [review]

**Pull request merged**